prboom2/src/p_pspr.c

```diff
@@ -229,10 +299,21 @@ int weapon_attack_alignment=0;
 
 int P_SwitchWeapon(player_t *player)
 {
-  int *prefer = weapon_preferences[demo_compatibility!=0]; // killough 3/22/98
-  int currentweapon = player->readyweapon;
-  int newweapon = currentweapon;
-  int i = NUMWEAPONS+1;   // killough 5/2/98
+  int *prefer;
+  int currentweapon, newweapon;
+  int i;
+
+  prefer = weapon_preferences[demo_compatibility != 0]; // killough 3/22/98
```

Comment:
Let's move these lines past the branch condition so we can save precious picoseconds 🙏

Reply:
I took care of it 🎵